pubtools/pulplib/_impl/fake/client.py

```diff
@@ -491,17 +491,16 @@ def _do_unassociate(self, repo_id, criteria=None):
         removed_units = set()
         kept_keys = set()
 
+        criteria = criteria or Criteria.true()
+        # only validating the criteria here like in actual scenario.
         # use type hint=Unit so that if type_ids are the goal here
```

Comment:
I think this old comment is obsolete now, isn't it?

It was saying that we call search_for_criteria just to get a pulp_search for which we can look at type_ids.  But your new comment says we are just doing this to validate the criteria object, which I think is valid, meaning the old comment can be removed.

Reply:
Agree. Updated the comment.